python-frontend/src/main/java/org/sonar/python/semantic/SymbolUtils.java

```diff
@@ -77,7 +77,7 @@ public static String fullyQualifiedModuleName(String packageName, String fileNam
     String moduleName = extensionIndex > 0
       ? fileName.substring(0, extensionIndex)
       : fileName;
-    if (moduleName.equals("__init__")) {
+    if (moduleName.equals("__init__") || packageName.endsWith("." + moduleName) || packageName.equals(moduleName)) {
```

Comment:
I don't fully understand the reason behind those extra conditions.
I see that you want to simplify logic inside `SymbolTableBuilder` and `Typeshed`, however this is not what I would expect from this function, i.e. if I call `fullyQualifiedModuleName('mod', 'mod.py')` I would expect the fully qualified name to be `mod.mod`

Here you have an example in python repl:
https://repl.it/repls/LastJauntyGlueware

Reply:
You are right. Got rid of this extra.